docs/frontend-system/architecture/05-extension-overrides.md

```diff
@@ -10,38 +10,107 @@ description: Frontend extension overrides
+- It is already possible to enable an extension that is already provided by the plugin. Imagine we have a Search plugin that provides a default search result item extension for rendering search results. Consider also that we have a Catalog plugin with a disabled Catalog search result item extension. To change the way Catalog search result items are rendered, you don't need to override the default result item extension. The only thing you need to do is enable the Catalog search result extension via the config file. For other types of search results, the default Search result item will continue to be used.
 
-## Creating a Extension Override
+## Creating an Extension Override
 
-<!--
+The following steps should be followed to override extensions:
 
-How to create an extension override and export it from a package.A
+```tsx
+// plugins/aperture-overrides/src/overrides.ts
+import {
+  createThemeApi,
+  createExtensionOverrides
+} from '@backstage/frontend-plugin-api';
 
- - Example using the default exports
+// Creating a light theme extension;
+const apertureLightTheme = createThemeApi({ … });
 
-You can also create them directly withing the app for local overrides.
+// Creating a light theme extension;
+const apertureDarkTheme = createThemeApi({ … });
 
- - Example of installing an override directly into an app
+// Creating an extension overrides preset
+export createExtensionOverrides({
+  Extensions: [apertureLightTheme, apertureDarkTheme]
+});
 
-Mention that in can still be a good idea to split your overrides out into separate packages in large projects. But it's up to you to decide how to group the extensions into extension overrides.
+// plugins/aperture-overrides/src/index.ts
+export { default } from './overrides';
+```
 
- -->
+For this example, we are creating overrides for the light and dark theme extensions and exporting the overrides from the plugin index file. Now we are able to use the overrides in a Backstage app:
 
-## Overriding Existing Extensions
+```tsx
+// packages/app/src/App.tsx
+import { createApp } from '@backstage/frontend-app-api';
+import apertureOverrides from ‘@backstage/plugin-aperture-overrides’
+
+const app = createApp({
+ features: [ apertureOverrides ],
+});
+
+export default app.createRoot().
+```
+
+If the plugin you want to change is internal to your company or you just want to replace one of the application's core extensions, you can decide to create replacements directly in the application for local replacements. See an example below:
+
+```tsx
+// packages/app/src/themes.ts
+import { createExtensionOverrides } from from '@backstage/frontend-plugin-api';
+
+// Creating the light theme extension;
+export const apertureLightTheme = createThemeApi({ … });
+
+// Creating the light theme extension;
+const apertureDarkTheme = createThemeApi({ … });
 
-<!--
+// Exporting your custom extensions
+export default [apertureLightTheme, apertureDarkTheme];
```

Comment:
I'm thinking we could just reference the package name that we export the override from in the first example instead? There's quite a lot of duplicate code here and I think we should encourage having overrides separated out into a different package.

Reply:
I replaced it with a single app example, is it better?